### PR TITLE
DTSRD-1176.provision terraform v14 server

### PR DESCRIPTION
**Pull request by SabinaHMCTS**
### JIRA link (if applicable) ###

DTSRD-1176

### Change description ###

provision terraform v14 server

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ *] No
```

### CODE REVIEW

**Review by lukasz-wolski — APPROVED** [review]

File: infrastructure/main.tf [hunk 1]
```diff
@@ -18,11 +26,6 @@ module "db-common-data-v11" {
   postgresql_version = "11"
 }
 
-data "azurerm_key_vault" "rd_key_vault" {
-  name                = local.key_vault_name
```
Comment (kiran-yenigala-hmcts) [inline infrastructure/main.tf:22]: why this is removed?

File: infrastructure/main.tf [hunk 2]
```diff
@@ -10,6 +17,7 @@ module "db-common-data-v11" {
   component          = var.component
   name               = join("-", [var.product, var.component, "postgres-db", "v11"])
   location           = var.location
+
```
Comment (kiran-yenigala-hmcts) [inline infrastructure/main.tf:20]: remove empty line

File: infrastructure/variables.tf [hunk 3]
```diff
@@ -4,15 +4,21 @@ variable "product" {
...
+  default="postgres-db-v14"
+}
 variable "location" {
   type = string
   default = "UK South"
 }
 
 variable "env" {
   type = string
+  default="demo"
```
Comment (kiran-yenigala-hmcts) [inline infrastructure/variables.tf:21]: why it is defined here?

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/util/CommonDataApiPostgresqlContainer.java [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 @Testcontainers
 public class CommonDataApiPostgresqlContainer extends PostgreSQLContainer<CommonDataApiPostgresqlContainer> {
-    private static final String IMAGE_VERSION = "postgres:11.1";
+    private static final String IMAGE_VERSION = "postgres:14";
```
Comment (kiran-yenigala-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/util/CommonDataApiPostgresqlContainer.java:9]: we can do this later, as we have to run the integrationTests against 11.1 for now?

File: src/integrationTest/resources/application-test.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 flaglist: PF0015,RA0042
 spring:
   datasource:
-    url: ${DB_URL:jdbc:tc:postgresql:11.1:///dbcommondata}
+    url: ${DB_URL:jdbc:tc:postgresql:14:///dbcommondata}
```
Comment (kiran-yenigala-hmcts) [inline src/integrationTest/resources/application-test.yml:4]: we can do this later, as we have to run the integrationTests against 11.1 for now?

File: infrastructure/variables.tf (hunk 3 above)
Comment (SabinaHMCTS) [reply infrastructure/variables.tf:21]: i thought for this demo branch i will default it  but i can keep it a s a variable

File: src/integrationTest/resources/application-test.yml (hunk 5 above)
Comment (SabinaHMCTS) [reply src/integrationTest/resources/application-test.yml]: reverted

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/util/CommonDataApiPostgresqlContainer.java (hunk 4 above)
Comment (SabinaHMCTS) [reply src/integrationTest/java/uk/gov/hmcts/reform/cdapi/util/CommonDataApiPostgresqlContainer.java]: reverted

File: infrastructure/main.tf (hunk 2 above)
Comment (SabinaHMCTS) [reply infrastructure/main.tf]: done

File: charts/rd-commondata-api/values.yaml [hunk 6]
```diff
@@ -6,8 +6,8 @@ java:
   image: 'hmctspublic.azurecr.io/rd/commondata-api:latest'
 
   environment:
-    POSTGRES_HOST: rd-commondata-api-postgres-db-v11-{{ .Values.global.environment }}.postgres.database.azure.com
-    POSTGRES_USERNAME: dbcommondata@rd-commondata-api-postgres-db-v11-{{ .Values.global.environment }}
+    POSTGRES_HOST: rd-commondata-api-postgres-db-v14-{{ .Values.global.environment }}.postgres.database.azure.com
+    POSTGRES_USERNAME: dbcommondata@rd-commondata-api-postgres-db-v14-{{ .Values.global.environment }}
```
Comment (kiran-yenigala-hmcts) [inline charts/rd-commondata-api/values.yaml:10]: we don't need to change these at the moment

Comment (SabinaHMCTS) [reply charts/rd-commondata-api/values.yaml]: reverted

**Pull request merged**